docs/quickstart/bootstrap.rst

```diff
@@ -104,6 +104,12 @@ Bootstrap node.
 
    root@bootstrap $ /srv/scality/metalk8s-|release|/bootstrap.sh
 
+.. warning::
+
+    In case of virtual network (or any network not able to route
```

Comment:
Not sure if _routing_ is the issue, since this is managed by Calico. The issue is that Calico is not encapsulating packets within a single network, leaving the origin and destination to internal IP addresses (each node being configured to know where to route packets).

Most virtual routers (such as OpenStack Neutron, see [here](https://docs.openstack.org/dragonflow/latest/specs/mac_spoofing.html)) enforce that all traffic within VMs can only contain packets with a `src` matching their actual MAC-IP pair on the virtual port (interface):

> Without this protection a VM in a network can spoof and answer to ARP requests that don’t actually belong to it, or fake response for other VM.

Not sure how we want to translate this into documentation, but it might be worth the effort (since we'll make this configurable in the future, see #1234).

Reply:
It's not only for virtual networks it's for all networks not able to route packet targetting internal IP.

If for what ever reason (firewall, some internal router in the network or .... what ever), a server or router or switch is not able to determine where he need to route this internal IP we need to have IP-in-IP encapsulation (or adding some default routes somewhere but this solution is not on our side)

Reply:
I agree with @gdemonet, *routing* per se is not the issue.

> it's for all networks not able to route packet targetting internal IP.

Well, no, that's why we enable IPIP for *CrossSubnet* traffic by default.

I'm not sure what you mean with 'internal IP'. Do you mean PodIP? That's *not* how the routing works, as in, packets to PodIPs are not sent to any routers, they're sent immediately to the *machine* that's hosting said PodIP (because we're in-subnet).

I.e., if I have PodIP 1.0.0.1, and I'm in network 2.0.0.0/8, and the Pod is running on a host with IP 2.0.0.2, then if a Pod on another host wants to connect to this 1.0.0.1 Pod, there'll be a routing rule (not literally, Calico uses ranges, but bear with me) which says "1.0.0.1/32 via 2.0.0.2 dev ..." (and there's another entry saying "2.0.0.0/8 dev ... proto kernel scope link" or similar). So, if a packet is sent to 1.0.0.1, it will be sent to the *MAC* of the machine which ARP-responds to 2.0.0.2.
Basically, in this whole transmission, there's no real 'routing' involved in the sense that there's no host through which the packets flow which is not the destination *machine* of the packets, this is direct one-to-one communication.

Reply:
Ok got it